### PR TITLE
Add examples/documentation for the test framework.

### PR DESCRIPTION
**Pull request by ozevren**

### CODE REVIEW

File: tests/integration/examples/example_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		NewSuite("galley_conversion", m).
+
+		// Labels that apply to the whole suite can be specified here.
+		Label(label.Presubmit).
+
+		// You can restrict the execution of the whole suite to a particular environment. This restricts execution
+		// to the native environment.
+		RequireEnvironment(environment.Native).
+
+		// You can specify multiple setup functions that will be run as part of suite setup. setupFn will alwaus be called.
```
Comment (howardjohn) [inline tests/integration/examples/example_test.go]: typo: always

**Comment by ozevren** [comment]
I think the main question here is, how would you verify the behavior? It
looks like simply looking at xDS is not enough, you want to verify the
actual behavior. In essence, it sounds like you want to write a e2e-like
tests.

Some of the Pilot tests should be working this way already.

File: tests/integration/README.md [hunk 2]
```diff
@@ -40,64 +48,96 @@ $ go test ./... -istio.test.env native
...
+When executed a non-supported environment, these tests will be skipped. 
 
 When an environment is not specified, the tests will run against the local environment by default.
 
 #### Kubernetes Environment
 
-When running the tests against the Kubernetes environment, you will need to explicitly specify a kube config
-file for the cluster you need to use. **Be aware that any existing Istio deployment will be altered and/or
-removed**.
+When running the tests against the Kubernetes environment, you will need to provide a K8s cluster ot run the tests
```
Comment (howardjohn) [inline tests/integration/README.md]: typo: to

File: tests/integration/README.md [hunk 3]
```diff
@@ -40,64 +48,96 @@ $ go test ./... -istio.test.env native
...
+against. You can specify the kube config file that should be used to use for connecting to the cluster, through 
+command-line: 
 
 ```console
-$ go test ./...  -istio.test.env kube -istio.test.kube.config ~/.kube/config
+$ go test ./...  --istio.test.env kuber --istio.test.kube.config ~/.kube/config
```
Comment (howardjohn) [inline tests/integration/README.md]: typo kuber

**Pull request merged**